gradle/runtime-jdk-provision.gradle

```diff
@@ -0,0 +1,25 @@
+import org.elasticsearch.gradle.OS
```

Comment:
Why a script plugin rather than putting this in `BuildPlugin` with all the rest of our test config? I get the motivation to break up `BuildPlugin` but we want to do so in the form of more binary plugins generally. This just naturally fits in the existing `configureTestTasks()` method and will make it a bit easier to refactor all that out into a separate binary plugin if this lives there too.

Reply:
I didn't want to add this to build-tools on purpose as it can get used and become something we'll have to support, or break users of it outside our build, it didn't seem to fit there, A script seems more appropriate to specifying a conventions that spans all the projects in _our_ build.

Reply:
Yeah, I think we need a better long term solution there. I don't think we have a good distinction of what conventions should apply to external users and what applies to the ES project. This is simply part of the fact that external use of build-tools is still very sketchy. I'm good with leaving this as-is for now. I agree, this isn't really applicable to external users.